admin/bin/check-ps-report-for-corrupt-data.js

```diff
@@ -0,0 +1,380 @@
+#!/usr/bin/env node
+'use strict'
+const fs = require('fs')
+const csv = require('fast-csv')
+const startTime = Date.now()
+const R = require('ramda')
+const RA = require('ramda-adjunct')
+
+// const timeRegex =  /^\d{1,2}:\d{2}:\d{2} (AM|PM)$/i // Not quite right - PM instead pm and single digit hours
+const timeRegex = /^\d{2}:\d{2}:\d{2} (AM|PM)$/ // Correct
+
+// Check for corrupt data in the PS report
+function help () {
+  console.log(`usage: ${process.argv[0]} <ps-file-to-inspect>`)
+}
+
+function report (prop, row, val) {
+  // console.log()
+  console.log(`${prop} field not matched for ${row['AttemptId']} val was '${val}'`)
+}
+
+function reportMissing (row, prop) {
+  process.stderr.write(`${prop} field missing for ${row['AttemptId']}\n`)
+}
+
+function reportProgress (count) {
+  // process.stdout.write('.')
+}
+
+function regexTest (row, prop, regex) {
+  if (row.hasOwnProperty(prop)) {
+    const val = row[prop]
+    if (val.length && !regex.test(val)) {
+      report(prop, row, val)
+    }
+  } else {
+    console.log(`Missing property: ${row['checkCode']} property: ${prop}`)
+  }
+}
+
+const checkID = function (row, prop) {
+  const regex = /^\d{1,2} x \d{1,2}$/
+  regexTest(row, prop, regex)
+}
+
+const checkResponse = function (row, prop) {
+  const regex = /^\d{0,5}$/
+  regexTest(row, prop, regex)
+}
+
+const checkK = function (row, prop) {
+  const val = row[prop]
+  const regex = /^(k|m|t|x)?\[(.*)\]*$/
+  const inputs = val.split(', ')
+  inputs.forEach(input => {
+    const s = input.trim()
+    if (s !== '' && !regex.test(s)) {
+      report(prop, row, val)
+    }
+  })
+}
+
+const checkSco = function (row, prop) {
+  const regex = /^(1|0)$/
+  regexTest(row, prop, regex)
+}
+
+function checkResponseTime (row, prop) {
+  const val = Number(row[prop])
+  if (typeof val !== 'number') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOut (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutResponse (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checkTimeOutSco (row, prop) {
+  const val = row[prop]
+  if (val !== '' && val !== '1' && val !== '0') {
+    report(prop, row, val)
+  }
+}
+
+function checktLoad (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktFirstKey (row, prop) {
+  regexTest(row, prop, timeRegex)
+}
+
+function checktLastKey (row, prop) {
```

Comment:
checkt?

Reply:
Field is called QntLastKey in the spec, so it becoms checktLastKey here.
Happy?

Reply:
all good 👍